### PR TITLE
Fix crashes on cypress realworld app

### PR DESCRIPTION
**Pull request by bhackett1024**
This PR has fixes for https://linear.app/replay/issue/RUN-1123 and https://linear.app/replay/issue/RUN-1127, which are the main issues causing crashes when replaying recordings from this test suite.

With these fixes, we were able to run the test suite twice without errors:

https://github.com/replayio/build-test-dashboard/blob/master/49d2fb62-c511-4aed-9ca3-9d52a77d360a

### CODE REVIEW

File: base/task/sequence_manager/task_queue_impl.cc [hunk 1]
```diff
@@ -1346,7 +1346,7 @@ void TaskQueueImpl::SetThrottler(TaskQueue::Throttler* throttler) {
 
 void TaskQueueImpl::ResetThrottler() {
   main_thread_only().throttler = nullptr;
-  LazyNow lazy_now(sequence_manager_->main_thread_clock());
+  LazyNow lazy_now(sequence_manager_->any_thread_clock_maybe_events_disallowed());
```
Comment (Domiii) [inline base/task/sequence_manager/task_queue_impl.cc:1349]: Why did we change it from `main` to `any`? 
(I assume, they are quite interchangeable?)

File: third_party/blink/renderer/core/messaging/message_port.cc [hunk 2]
```diff
@@ -177,16 +177,16 @@ void MessagePort::start() {
...
   // A closed port should not be neutered, so rather than merely disconnecting
   // from the mojo message pipe, also entangle with a new dangling message pipe.
   if (!IsNeutered()) {
     // Refuse to entangle the port at non-deterministic points, as this requires
-    // creating mojo resources.
-    if (recordreplay::AreEventsDisallowed())
+    // creating mojo resources. close() calls are always treated as non-deterministic
+    // because the port's lifetime is managed by the GC and on destruction the
+    // port is detached from the connector.
+    if (recordreplay::IsRecordingOrReplaying("disallow-events"))
```
Comment (Domiii) [inline third_party/blink/renderer/core/messaging/message_port.cc:189]: 1) Absolutely awesome explanation. I am learning a lot here!

2) However, is there a chance that this is causing some of the recorder crashes in `MessagePort::Accept`?
E.g.: https://ui.honeycomb.io/replay/datasets/backend/result/EygotZ8iRif/a/ovMwSPjA2mG/Recorder-Crash-Reports?tab=raw

File: base/task/sequence_manager/task_queue_impl.cc (hunk 1 above)
Comment (bhackett1024) [reply base/task/sequence_manager/task_queue_impl.cc:1349]: This is for simplicity, the any_thread version subsumes the main_thread version as they're both atomic accesses with different memory order constraints.

File: third_party/blink/renderer/core/messaging/message_port.cc (hunk 2 above)
Comment (bhackett1024) [reply third_party/blink/renderer/core/messaging/message_port.cc:189]: It's possible, yeah.  The best thing in this case is to file an issue in linear, though, not a github comment.

Comment (bhackett1024) [reply third_party/blink/renderer/core/messaging/message_port.cc:189]: https://linear.app/replay/issue/RUN-1176

**Pull request merged**